src/hotspot/share/memory/arena.hpp

```diff
@@ -32,10 +32,8 @@
 
 #include <new>
 
-// The byte alignment to be used by Arena::Amalloc.  See bugid 4169348.
-// Note: this value must be a power of 2
-
-#define ARENA_AMALLOC_ALIGNMENT (2*BytesPerWord)
+// The byte alignment to be used by Arena::Amalloc.
+#define ARENA_AMALLOC_ALIGNMENT BytesPerLong
```

Comment:
We don't store types in here which need alignment larger than 64bit? eg long double?

.. I searched and found no cases where they lived inside an arena, so it's probably fine.

Reply:
I don't think so.  I'm not sure how to search for that if there are any.